c7n/ufuncs/s3crypt.py

```diff
@@ -48,37 +48,51 @@ def process_key_event(event, context):
         key = {'Key': record['s3']['object']['key'],
                'Size': record['s3']['object']['size']}
         version = record['s3']['object'].get('versionId')
+        if version is not None:
+            key['VersionId'] = version
+            # lambda event is always latest version, but IsLatest
+            # is not in record
+            key['IsLatest'] = True
+            method = processor.process_version
+        else:
+            method = processor.process_key
         try:
-            if version is not None:
-                key['VersionId'] = version
-                # lambda event is always latest version, but IsLatest
-                # is not in record
-                key['IsLatest'] = True
-                result = retry(processor.process_version, s3, key, bucket)
-            else:
-                result = retry(processor.process_key, s3, key, bucket)
+            result = retry(method, s3, key, bucket)
         except ClientError as e:
             # Ensure we know which key caused an issue
             print("error %s:%s code:%s" % (
                 bucket, key['Key'], e.response['Error']))
-            raise
+            print("trying with urldecoded key")
+            try:
+                decode = urllib.unquote(key['Key'])
```

Comment:
The reason for the pull request is that a file in S3 that is saved as something like `5:00.log` will be returned in the JSON event as `5%3A00.log`.

The reason it is not urldecoded unconditionally is because some people may actually be saving the file as `5%3A00.log`